addons/product_visible_discount/product_visible_discount.py

```diff
@@ -80,29 +101,36 @@ def get_real_price(res_dict, product_id, qty, uom, pricelist):
         result=res['value']
         pricelist_obj=self.pool.get('product.pricelist')
         product_obj = self.pool.get('product.product')
+
         if product and pricelist and self.pool.get('res.users').has_group(cr, uid, 'sale.group_discount_per_so_line'):
-            if result.get('price_unit',False):
+            if result.get('price_unit',False) is not False:
```

Comment:
because we had to deal with gratis products and needed to discriminate cases where price_unit is unset vs set to 0.